Source/JavaScriptCore/runtime/JSArrayBufferConstructor.cpp

```diff
@@ -81,27 +81,51 @@ EncodedJSValue JSGenericArrayBufferConstructor<sharingMode>::constructImpl(JSGlo
     Structure* structure = JSC_GET_DERIVED_STRUCTURE(vm, arrayBufferStructureWithSharingMode<sharingMode>, newTarget, callFrame->jsCallee());
     RETURN_IF_EXCEPTION(scope, { });
 
-    size_t length;
+    size_t length = 0;
+    std::optional<size_t> maxByteLength;
     if (callFrame->argumentCount()) {
-        length = callFrame->uncheckedArgument(0).toTypedArrayIndex(globalObject, "length");
-        RETURN_IF_EXCEPTION(scope, encodedJSValue());
-    } else {
-        // Although the documentation doesn't say so, it is in fact correct to say
-        // "new ArrayBuffer()". The result is the same as allocating an array buffer
-        // with a zero length.
-        length = 0;
+        length = callFrame->uncheckedArgument(0).toTypedArrayIndex(globalObject, "length"_s);
+        RETURN_IF_EXCEPTION(scope, { });
+
+        if constexpr (sharingMode == ArrayBufferSharingMode::Shared) {
+            if (Options::useResizableArrayBuffer()) {
+                JSValue options = callFrame->argument(1);
```

Comment:
What ensures that we have at least 2 arguments on the stack?

Reply:
`argument(1)` checks it. `uncheckedArgument` does not check it. So if we do not have an argument, it is totally fine.